### PR TITLE
Fix - Code smell feast

### LINKED ISSUE #263: Project is still using JDK 8, Maven 3.3
Hi Raphaël, hope you're well,
as I understand the structure of the code, it's still using JDK 8/11 and Maven 3.3.3.

I have noticed a few other things:
- System.out calls
- java.util.logger that leaves little to no option for configuration
- lack of streams support

I am sure that we could benefit from refreshing the project a bit, up to a great bump in performance.

Would you allow such a refresh?
There would be backward compatibilities issues of course, but older users could still use versions up to 3.7.0.
Newer users would benefit from the bump-up, version 4.0.0 I guess (if you follow semantic versioning).

### PR DESCRIPTION
**Pull request by rdlopes**
This PR removes a few Sonar bugs and code smells related to:
- Java constructs
- Deprecations
- Java language migration
- Nullability issues
- API usage

I didn't touch on the deprecated API, but marked it for removal for next 4.0.0 version.

Closes #263 for the Java/Maven part.

### CODE REVIEW

File: cukedoctor-converter/src/main/java/com/github/cukedoctor/util/builder/FeatureBuilder.java [hunk 1]
```diff
@@ -81,146 +87,134 @@ public Feature build() {
 
   public Feature aFeatureWithTwoScenarios() {
```
Comment (rmpestano) [inline cukedoctor-converter/src/main/java/com/github/cukedoctor/util/builder/FeatureBuilder.java:88]: I can't remember why these `aFeatureWith*` are in main source, I think we should move to test folder

**Review by rmpestano — CHANGES_REQUESTED** [review]
Really fantastic work, thank you again for your contribution! I left some minor suggestions/questions

File: cukedoctor-converter/src/perf/java/com/github/cukedoctor/CukedoctorBenchmark.java [hunk 2]
```diff
@@ -24,62 +30,62 @@
...
+          build()
+      ).run();
+    } finally {
+      List<String> files = FileUtil.findFiles("target/benchmark", ASCIIDOC_FILE_EXTENSION);
+      log.info("Number of files converted: {}", files.size());
+      CukedoctorBenchmark.removeAdocFIles();
     }
+  }
+
+  private static void removeAdocFIles() {
```
[the remark this answers is not in the document]
Comment (rdlopes) [reply cukedoctor-converter/src/perf/java/com/github/cukedoctor/CukedoctorBenchmark.java:78]: Thanks!

File: cukedoctor-converter/src/test/java/com/github/cukedoctor/converter/CukedoctorConverterTest.java [hunk 3]
```diff
@@ -142,23 +147,16 @@ public void shouldNotSetInvalidFilename() {
...
+    assertThatThrownBy(() -> converter.setFilename("test.txt"))
+        .isInstanceOf(RuntimeException.class)
+        .hasMessage(
+            "Invalid filename extension for file: test.txt. Valid formats are: ad, adoc, asciidoc"
+                + " and asc");
+    assertThatThrownBy(() -> converter.setFilename("test.doc"))
+        .isInstanceOf(RuntimeException.class)
+        .hasMessage(
+            "Invalid filename extension for file: test.doc. Valid formats are: ad, adoc, asciidoc"
+                + " and asc");
```
[the remark this answers is not in the document]
Comment (rdlopes) [reply cukedoctor-converter/src/test/java/com/github/cukedoctor/converter/CukedoctorConverterTest.java:159]: 🤓

File: cukedoctor-converter/src/test/java/com/github/cukedoctor/renderer/CukedoctorSummaryRendererTest.java [hunk 4]
```diff
@@ -22,12 +25,6 @@ public void shouldRenderSummaryForOneFeature() {
     String resultDoc =
         new CukedoctorSummaryRenderer()
             .renderSummary(features, CukedoctorDocumentBuilder.Factory.newInstance());
-    assertThat(resultDoc)
-        .isNotNull()
-        .containsOnlyOnce("<<One-passing-scenario-one-failing-scenario>>")
-        .containsOnlyOnce("|[red]#*failed*#")
-        .contains("2+|010ms");
-
     assertThat(resultDoc).isEqualTo(Expectations.SUMMARY_FOR_ONE_FEATURE);
```
[the remark this answers is not in the document]
Comment (rdlopes) [reply cukedoctor-converter/src/test/java/com/github/cukedoctor/renderer/CukedoctorSummaryRendererTest.java:28]: 😉

File: cukedoctor-converter/src/test/resources/com/github/cukedoctor/bdd/cukedoctor/intro-chapter.feature [hunk 5]
```diff
@@ -37,109 +37,11 @@ Feature: Feature2
...
-
-==========
-Given ::
-scenario step icon:thumbs-up[role="green",title="Passed"] [small right]#(000ms)#
-==========
-
-
-"""
+    Then Resulting documentation should have the provided introduction chapter
+"""https://"""
```
[the remark this answers is not in the document]
Comment (rdlopes) [reply cukedoctor-converter/src/test/resources/com/github/cukedoctor/bdd/cukedoctor/intro-chapter.feature:47]: What a shame! I'm sorry I didn't see this, totally unintentional, good catch 😱

File: cukedoctor-extension/src/main/resources/themes/rocket-panda.css [hunk 6]
```diff
@@ -1,4 +1,4 @@
-@import url(http://red-hat-overpass-fonts.s3.amazonaws.com/overpass.css);
+@import url(https://);
```
[the remark this answers is not in the document]
Comment (rdlopes) [reply cukedoctor-extension/src/main/resources/themes/rocket-panda.css:1]: It's still a shame to have deleted this.
This change is for applying https schemas and got a bit heavy on the find/replace tool. Unacceptable, I'm putting it back for you to delete 👍

File: cukedoctor-extension/src/test/java/com/github/cukedoctor/extension/StyleExtensionTest.java [hunk 7]
```diff
@@ -9,9 +9,13 @@
...
 import org.asciidoctor.Asciidoctor;
-import org.asciidoctor.OptionsBuilder;
+import org.asciidoctor.Options;
 import org.asciidoctor.SafeMode;
-import org.junit.*;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
```
[the remark this answers is not in the document]
Comment (rdlopes) [reply cukedoctor-extension/src/test/java/com/github/cukedoctor/extension/StyleExtensionTest.java:18]: Nope, no option 🥹
Actually, it's Sonar which rants about wildcards in imports that are not static.
For the static imports, I guess we will still rely on google java format and IntelliJ IDE.

File: cukedoctor-converter/src/main/java/com/github/cukedoctor/util/builder/FeatureBuilder.java (hunk 1 above)
Comment (rdlopes) [reply cukedoctor-converter/src/main/java/com/github/cukedoctor/util/builder/FeatureBuilder.java]: Well spotted! Moved that to test scope in a utility class called Features

File: cukedoctor-converter/src/test/resources/com/github/cukedoctor/bdd/cukedoctor/intro-chapter.feature (hunk 5 above)
[the remark this answers is not in the document]
Comment (rmpestano) [reply cukedoctor-converter/src/test/resources/com/github/cukedoctor/bdd/cukedoctor/intro-chapter.feature]: No problem

File: cukedoctor-extension/src/main/resources/themes/rocket-panda.css (hunk 6 above)
[the remark this answers is not in the document]
Comment (rmpestano) [reply cukedoctor-extension/src/main/resources/themes/rocket-panda.css]: No worries 👍

**Review by rmpestano — APPROVED** [review]

**Pull request merged**